code/projects/app/src/app/shared/file-download/file-download.component.ts

```diff
@@ -1,29 +1,28 @@
+import { AsyncPipe } from "@angular/common";
 import { Component, Input, OnInit } from "@angular/core";
-import { Observable } from "rxjs";
+import { Observable, Subject } from "rxjs";
 
 type DownloadData = {
   fileName: string;
   path: string;
-} | null;
+};
 
 @Component({
   selector: "app-file-download",
   standalone: true,
-  imports: [],
+  imports: [AsyncPipe],
   templateUrl: "./file-download.component.html",
   styleUrl: "./file-download.component.css",
 })
 export class FileDownloadComponent implements OnInit {
   @Input({ required: true }) public file$!: Observable<File>;
 
-  protected data: DownloadData = null;
+  private download = new Subject<DownloadData>();
+  protected download$ = this.download.asObservable()
 
-  public ngOnInit(): void {
+  ngOnInit(): void {
     this.file$.subscribe((file) => {
-      this.data = {
-        path: URL.createObjectURL(file),
-        fileName: file.name
-      }
-    });
+      this.download.next({fileName: file.name, path: URL.createObjectURL(file)})
+    })
   }
```

Comment:
Just use `map`?

```
protected readonly download$ = this.file$.pipe(map(file => ({ fileName: file.name, path: URL.createObjectURL(file) })))
```

And then `download` (without `$`) seems completely unnecessary as well as the entire `ngOnInit`, since `AsyncPipe` does subscribe anyway.

I might be missing the point here since I've just woken up recently.